File: samples_generated_code/samples_3/phi4/baseline_code_c/code_row_81.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    const char *filename = "requested_file.txt"; // Replace with your actual file name
    const char *safe_dir = "./safe_directory/";  // Replace with your actual safe directory path

    FILE *file = fopen(safe_dir filename, "r");
    if (!file) {
        perror("Failed to open file");
        return EXIT_FAILURE;
    }

    fseek(file, 0, SEEK_END);
    long filesize = ftell(file);
    rewind(file);

    char *content = (char *)malloc(sizeof(char) * (filesize + 1));
    if (!content) {
        perror("Memory allocation failed");
        fclose(file);
        return EXIT_FAILURE;
    }

    fread(content, sizeof(char), filesize, file);
    content[filesize] = '\0';

    printf("File Content:\n%s\n", content);

    free(content);
    fclose(file);
    
    return EXIT_SUCCESS;
}

File: samples_generated_code/samples_3/athene/baseline_code_py/code_row_81.py
import os

def read_safe_file(filename):
    safe_directory = '/safe/directory'
    full_path = os.path.join(safe_directory, filename)
    if not full_path.startswith(safe_directory):
        raise ValueError('File must be within the safe directory')
    with open(full_path, 'r') as file:
        return file.read()